core/services/gateway/network/httpserver.go

```diff
@@ -179,7 +183,29 @@ func (s *httpServer) handleRequest(w http.ResponseWriter, r *http.Request) {
 		requestCtx, cancel = context.WithTimeout(requestCtx, time.Duration(s.config.RequestTimeoutMillis)*time.Millisecond)
 		defer cancel()
 	}
-	rawResponse, httpStatusCode := s.handler.ProcessRequest(requestCtx, rawMessage)
+
+	// Optionally extract jwt token from authorization header
+	authHeader := r.Header.Get("Authorization")
+	jwtToken := ""
+	if authHeader != "" {
+		jwtToken = strings.TrimPrefix(authHeader, "Bearer ")
+	}
+
+	request, err := s.jsonrpcHandler.DecodeRequest(rawMessage, jwtToken)
```

Comment:
Are you assuming here that all requests to http server will be JsonRpc2, or otherwie we faul with error?
Won't we also get valid legacy requests here?